packages/libs/eda/src/lib/map/analysis/hooks/standaloneMapMarkers.tsx

```diff
@@ -0,0 +1,604 @@
+    [markerData]
+  );
+
+  const defaultDependentAxisRange = useDefaultAxisRange(
+    null,
+    0,
+    valueMinPos,
+    valueMax,
+    dependentAxisLogScale
+  ) as NumberRange;
+
+  /**
+   * Merge the overlay data into the basicMarkerData, if available,
+   * and create markers.
+   */
+  const markers = useMemo(() => {
+    return markerData.value?.mapElements.map(
+      ({
+        geoAggregateValue,
+        entityCount,
+        avgLat,
+        avgLon,
+        minLat,
+        minLon,
+        maxLat,
+        maxLon,
+        overlayValues,
+      }) => {
+        const bounds = {
+          southWest: { lat: minLat, lng: minLon },
+          northEast: { lat: maxLat, lng: maxLon },
+        };
+        const position = { lat: avgLat, lng: avgLon };
+
+        const donutData =
+          vocabulary && overlayValues && overlayValues.length
+            ? overlayValues.map(({ binLabel, value }) => ({
+                label: binLabel,
+                value: value,
+                color:
+                  overlayType === 'categorical'
+                    ? ColorPaletteDefault[vocabulary.indexOf(binLabel)]
+                    : gradientSequentialColorscaleMap(
+                        vocabulary.indexOf(binLabel) / (vocabulary.length - 1)
+                      ),
+              }))
+            : [];
+
+        // TO DO: address diverging colorscale (especially if there are use-cases)
+
+        // now reorder the data, adding zeroes if necessary.
+        const reorderedData =
+          vocabulary != null
+            ? vocabulary.map(
+                (
+                  overlayLabel // overlay label can be 'female' or a bin label '(0,100]'
+                ) =>
+                  donutData.find(({ label }) => label === overlayLabel) ?? {
+                    label: fixLabelForOtherValues(overlayLabel),
+                    value: 0,
+                  }
+              )
+            : // however, if there is no overlay data
+              // provide a simple entity count marker in the palette's first colour
+              [
+                {
+                  label: 'unknown',
+                  value: entityCount,
+                  color: '#333',
+                },
+              ];
+
+        // TO DO: this will sum to 1 when we allow proportion mode
+        // maybe ask for the counts to come along site the values in the back end response
+        // https://github.com/VEuPathDB/EdaDataService/issues/261
+        // (the way we did it in the old viz wasn't 100% accurate)
+        const count = reorderedData.reduce(
+          (sum, item) => (sum = sum + item.value),
+          0
+        );
+
+        const commonMarkerProps = {
+          id: geoAggregateValue,
+          key: geoAggregateValue,
+          bounds: bounds,
+          position: position,
+          data: reorderedData,
+          duration: defaultAnimationDuration,
+        };
+
+        switch (markerType) {
```

Comment:
Here's the switch - your comment seemed to suggest that we might leave this to a future ticket?

Reply:
Oh, great! Yes, merging sounds good. Maybe give @adnauseum a head's up, since this could impact his work on #111. I foresee tricky merge conflicts.